src/qililab/automatic_calibration/calibration_controller.py

```diff
@@ -0,0 +1,476 @@
+# Copyright 2023 Qilimanjaro Quantum Tech
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Automatic-calibration Controller module, which works with notebooks as nodes."""
+from datetime import datetime, timedelta
+
+import networkx as nx
+
+from qililab.automatic_calibration.calibration_node import CalibrationNode
+from qililab.data_management import build_platform, save_platform
+from qililab.platform.platform import Platform
+
+
+class CalibrationController:
+    """Class that controls the automatic calibration sequence.
+
+    Using this class, you normally would run:
+        - ``CalibrationController.run_automatic_calibration()`` for calibrating the full graph.
+        - ``CalibrationController.maintain(node)`` for targeting and making sure a node works.
+
+    |
+
+    **The calibration workflow, has three levels of methods that manage it:**
+
+    The highest level method to apply, is ``run_automatic_calibration()`` which finds all the final nodes (don't have others past them), and runs ``CalibrationController.maintain()`` on those.
```

Comment:
It's tricky, becuase the auto-calibration notation used in papers has the arrows in the other way to what is intuitive!

So in reality the right nodes, are the roots and the left ones the leaves since the arrows go from right to left! (direction of Ramiro autocalibration schema for example, or the graphs I draw in the tests.)

But yes surely we can talk about which convention we want for this, I think it would be a good discussion to have with the team!

Reply:
I am fine with this explanation, but it is worthy to express it in documentation then. If we are following that nomenclature, we need to detail it so it is clear to users and developers.

Reply:
Totally agree! I'm thinking where to place it and if to use that format or the opposite one!

Reply:
After thinking about this, it didn;t really make sense to follow google paper on this, since it makes no sense from the user point of view!

Also google paper is from 2018, if you check at their more recent images, from the Youtube video for example, yu can see they have changed it, to what makes sense:
![image](https://github.com/qilimanjaro-tech/qililab/assets/109400222/fc0e80ef-d676-4282-b3ba-dcc7d4984f57)
![image](https://github.com/qilimanjaro-tech/qililab/assets/109400222/e1313910-fb9d-4f1a-9f12-ca2866914be4)
![image](https://github.com/qilimanjaro-tech/qililab/assets/109400222/7c833752-d671-45bb-81e3-9bc66669023a)

In our hearths, Isaac and I really knew we needed to change this, and with this last insight, I just changed it!

Also documented it better in the class, so there are no doubts! 🙌